cpp/src/slice2cs/Gen.cpp

```diff
@@ -193,6 +197,28 @@ Slice::CsVisitor::writeMarshal(const OperationPtr& operation, bool returnType)
                                    members.size() == 1 ? "value" : "value." + fieldName(member), member->tag());
         }
 
+        if (members.back()->stream())
+        {
+            if (i == 0)
+            {
+                if (members.size() == 1)
+                {
+                    _out << nl << "return socketStream => socketStream." << streamDataWriter(members.back()->type());
+                    _out << "(value, cancel);";
+                }
+                else
+                {
+                    _out << nl << "var streamableValue = value." + fieldName(members.back()) << ";";
+                }
+                _out << nl << "return socketStream => socketStream." << streamDataWriter(members.back()->type());
+                _out << "(" << (members.size() == 1 ? "value" : "streamableValue") << ", cancel);";
+            }
+            else
+            {
+                _out << nl << "return null;";
```

Comment:
Ok I'll remove the loop check and generate the same code for both 1.1 and non-1.1 encodings. For 1.1, the code is never reached because there's no data frames so ReadReturnValue raises InvalidDataException. Another option could be to return an empty IO stream as well.